pydantic/networks.py

```diff
@@ -185,9 +185,14 @@ def validate(cls, value: Any, field: 'ModelField', config: 'BaseConfig') -> 'Any
         scheme = parts['scheme']
         if scheme is None:
             raise errors.UrlSchemeError()
+
         if cls.allowed_schemes and scheme.lower() not in cls.allowed_schemes:
             raise errors.UrlSchemePermittedError(cls.allowed_schemes)
 
+        port = parts['port']
+        if port is not None and int(port) > 65_536:
```

Comment:
```suggestion
        if port is not None and int(port) > 65_535:
```
You allow `65536` here !
But I think it would make more sense to keep `2 ** 16 - 1` to keep the logic behind

Reply:
oh yes it was a typo,
I thought it would avoid computation but the the difference is really insignifiant
```py
>>> import timeit
>>> def f1():
	return 65_535

>>> timeit.timeit(f1)
0.0505603000000292
>>> def f2():
	return 2**16-1

>>> timeit.timeit(f2)
0.05037179999999353
>>> def f3():
	return (1<<16)-1

>>> timeit.timeit(f3)
0.051782900000034715
```